File: languages/go/exercises/concept/basics/basics_test.go
package basics

import (
	"testing"
)

type lasagnaTests struct {
	layers, time, expected int
}

func TestOvenTime(t *testing.T) {
	got := OvenTime()
	if got != 40 {
		t.Errorf("OvenTime(40)  = %d; want 40", got)
	}
}

func TestRemainingOvenTime(t *testing.T) {
	got := RemainingOvenTime(25)
	if got != 15 {
		t.Errorf("RemainingOvenTime(25) = %d; want 15", got)
	}
}
func TestPreparationTime(t *testing.T) {
	var tests = []lasagnaTests{
		{ //one layer
			layers:   1,
			expected: 2,
		},
		{ //multiple layers
			layers:   4,
			expected: 8,
		},
	}

	for _, tc := range tests {
		got := PreparationTime(tc.layers)
		if got != tc.expected {
			t.Errorf("PreparationTime(%d) = %d; want %d", tc.layers, got, tc.expected)
		}
	}
}

func TestTotalTime(t *testing.T) {
	var tests = []lasagnaTests{
		{ //one layer
			layers:   1,
			time:     30,
			expected: 32,
		},
		{ //multiple layers
			layers:   4,
			time:     8,
			expected: 16,
		},
	}
	for _, tc := range tests {
		got := TotalTime(tc.layers, tc.time)
		if got != tc.expected {
			t.Errorf("PreparationTime(%d, %d) = %d; want %d", tc.layers, tc.time, got, tc.expected)
		}
	}
}

File: languages/cpp/exercises/concept/basics/basics_test.cpp
#include "basics.h"
#include "test/catch.hpp"

using namespace std;

TEST_CASE("Expected minutes in oven") {
  const int actual = Lasagna().expectedMinutesInOven();
  const int expected = 40;

  REQUIRE(expected == actual);
}

#if defined(EXERCISM_RUN_ALL_TESTS)
TEST_CASE("Remaining minutes in oven") {
  const int actual = Lasagna().remainingMinutesInOven(25);

  const int expected = 15;

  REQUIRE(actual == expected);
}

TEST_CASE("Preparation time in minutes for one layer") {
  const int actual = Lasagna().preparationTimeInMinutes(1);

  const int expected = 2;

  REQUIRE(actual == expected);
}

TEST_CASE("Preparation time in minutes for multiple layers") {
  const int actual = Lasagna().preparationTimeInMinutes(4);

  const int expected = 8;

  REQUIRE(actual == expected);
}

TEST_CASE("Elapsed time in minutes for one layer") {
  const int actual = Lasagna().elapsedTimeInMinutes(1, 30);

  const int expected = 32;

  REQUIRE(actual == expected);
}

TEST_CASE("Elapsed time in minutes for multiple layers") {
  const int actual = Lasagna().elapsedTimeInMinutes(4, 8);

  const int expected = 16;

  REQUIRE(actual == expected);
}
#endif
